### PR TITLE
Update dependencies to latest versions.

### PR DESCRIPTION
**Pull request by dmytro46**

### CODE REVIEW

File: lib/src/optional/default_optional_update_service.dart [hunk 1]
```diff
@@ -26,13 +26,14 @@ class DefaultOptionalUpdateService implements OptionalUpdateService {
...
-        final iTunesResults = await ITunesSearchAPI().lookupById(
+        final iTunesSearchAPI = ITunesSearchAPI();
+        final iTunesResults = await iTunesSearchAPI.lookupById(
           updateConfig.appStoreAppId!,
           country: updateConfig.appstoreCountryCode,
         );
         if (iTunesResults != null) {
           // Parse AppStore response
-          final iTunesVersionString = ITunesResults.version(iTunesResults);
+          final iTunesVersionString = iTunesSearchAPI.version(iTunesResults);
```
Comment (JTorrus) [inline lib/src/optional/default_optional_update_service.dart:36]: Shouldn't this be `iTunesResults` instead?

**Review by JTorrus — COMMENTED** [review]
Looks good! Left a comment to review before approving

File: lib/src/optional/default_optional_update_service.dart (hunk 1 above)
Comment (dmytro46) [reply lib/src/optional/default_optional_update_service.dart:36]: @JTorrus `iTunesSearchAPI.version()` this method was static in old version of `upgrader` package but now ITunesResults is `extension ITunesResults on ITunesSearchAPI ` and `version()` is a member method of `ITunesSearchAPI`

File: pubspec.yaml [hunk 2]
```diff
@@ -1,27 +1,28 @@
 name: app_versioning
 description: App versioning package
-version: 1.0.4
+version: 1.0.5
 publish_to: none
 
 environment:
-  sdk: ">=2.12.0 <3.0.0"
+  sdk: ">=2.12.0 <4.0.0"
+  flutter: ">=3.10.0" # upgrader package requires this flutter min version
```
Comment (Alqueraf) [inline pubspec.yaml:8]: Since we are forced to use flutter `3.10.0` and higher we can also enforce the dark sdk v3 and higher:
`sdk: ">=3.0.0 <4.0.0"`

**Review by Alqueraf — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**